File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

int N, M, V;
vector<vector<int>> adj;
vector<bool> visited;
vector<bool> visited2;
vector<int> ans1;
vector<int> ans2;

void dfs(int m) {
    visited[m] = true;
    ans1.push_back(m);
    for (int i : adj[m]) {
        if (!visited[i]) {
            dfs(i);
        }
    }
}

void bfs(int m) {
    queue<int> q;
    q.push(m);
    visited2[m] = true;
    ans2.push_back(m);

    while (!q.empty()) {
        int cur = q.front();
        q.pop();

        for (int i : adj[cur]) {
            if (!visited2[i]) {
                visited2[i] = true;
                ans2.push_back(i);
                q.push(i);
            }
        }
    }
}

int main() {
    cin >> N >> M >> V;

    adj.resize(N + 1);
    visited.resize(N + 1, false);
    visited2.resize(N + 1, false);

    for (int i = 0; i < M; ++i) {
        int a, b;
        cin >> a >> b;
        adj[a].push_back(b);
        adj[b].push_back(a);
    }

    for (int i = 1; i <= N; ++i) {
        sort(adj[i].begin(), adj[i].end());
    }

    dfs(V);
    for (int i = 0; i < ans1.size(); ++i) {
        cout << ans1[i];
        if (i != ans1.size() - 1) cout << " ";
    }
    cout << "\n";

    bfs(V);
    for (int i = 0; i < ans2.size(); ++i) {
        cout << ans2[i];
        if (i != ans2.size() - 1) cout << " ";
    }
    cout << "\n";

    return 0;
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque


N, M, V = map(int, input().split(' '))

adj = [[] for i in range(N+1)]

for i in range(M):
    a, b = map(int, input().split(' '))
    adj[a].append(b)
    adj[b].append(a)

for i in range(len(adj)):
    adj[i].sort()


ans1 = ''
visited = [0] * (N+1)

def dfs(m):
    global ans1
    if not visited[m]:
        visited[m] = 1
        ans1 += f'{m} '
        for i in adj[m]:
            dfs(i)

dfs(V)
print(ans1)

ans2 = ''  
visited2 = [0] * (N+1)



def bfs(m):

    global ans2 
    que = deque([m])
    visited2[m] = 1
    ans2 += f'{m} '
    
    while que:
        pop_num = que.popleft()
        for num in adj[pop_num]:
            if not visited2[num]:
                visited2[num] = 1
                ans2 += f'{num} '
                que.append(num)
    
bfs(V)
print(ans2)
